pkg/v2/service/filter/validation.go

```diff
@@ -170,6 +171,10 @@ func (f *validationPropertyFilter) validateUniqueness(ctx context.Context, nav p
 	if err != nil {
 		return err
 	} else if n > 0 {
+		if property.Attribute().Path() == "userName" {
+			return fmt.Errorf("%w: value of '%s' is not unique", spec.ErrUniqueness, property.Attribute().Path())
+		}
+
```

Comment:
There is no need to check for the path name here. Actually, checking for path names should be avoided in general as it breaks extensibility of the core module.

We can just change the line underneath to return an `spec.ErrUniqueness` error.

I will merge this in to `fix/65` branch and update it from there.